### PR TITLE
Implement load_extensions

### PR DESCRIPTION
**Pull request by thesadru**
## Summary

Add a method that loads all extensions in a directory.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `pre-commit run --all-files`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by rf20008 — APPROVED** [review]
Haven't tested the code, but this looks good.

Nitpick: RST-style docstrings for the functions. 

Otherwise, it's good :-)

File: disnake/ext/commands/common_bot_base.py [hunk 1]
```diff
@@ -591,6 +591,11 @@ def reload_extension(self, name: str, *, package: Optional[str] = None) -> None:
             sys.modules.update(modules)
             raise
 
+    def load_extensions(self, path: str) -> None:
+        """Loads all extensions in a directory."""
+        for extension in disnake.utils.search_directory(path):
+            self.load_extension(extension)
+
```
Comment (m1raynee) [inline disnake/ext/commands/common_bot_base.py:598]: some people might want to filter out not extensions
```suggestion
    def load_extensions(self, path: str, *, key: Callable[[str], bool] = None) -> None:
        """Loads all extensions in a directory."""
        if key is None:
            key = lambda: True
        for extension in disnake.utils.search_directory(path):
            if key(extension):
                self.load_extension(extension)
```

Comment (thesadru) [reply disnake/ext/commands/common_bot_base.py:598]: In that case you should probably be using `search_directories` directly instead.

**Comment by EQUENOS** [comment]
Looks good! Same nitpick: add docs for parameters in both functions to keep consistency with the rest of the documentation.

**Comment by shiftinv** [comment]
The docs of `Bot.load_extensions` should mention that it only loads *modules* from the specified directory, not necessarily all extensions. Assuming one has files like `cogs/test.py`, `cogs/core.py` etc., running `.load_extensions('cogs')` wouldn't load anything (maybe it should? idk).

`utils.search_directory` doesn't show up in docs, it would need to be added to `api.rst`

**Comment by shiftinv** [comment]
Oh wait, nevermind, I just didn't use it correctly. I used a relative path (`./cogs`) from `bot/__main__.py`, it should've been `./bot/cogs`.

**Review by shiftinv — APPROVED** [review]

**Pull request merged**